util/HalideTraceViz.cpp

```diff
@@ -789,9 +794,11 @@ int run(int argc, char **argv) {
         case halide_trace_end_produce:
         case halide_trace_consume:
         case halide_trace_end_consume:
+            break;
         case halide_trace_begin_pipeline:
         case halide_trace_end_pipeline:
-            break;
+            std::cerr << "Misplaced tracing event: " << p.event << "\n";
```

Comment:
I'm not so sure about this change - you get pipeline begin/end events when you trace something that has extern stages that are also Halide-being-traced.

Reply:
so nested begin/end pipeline events should just be ignored?

Reply:
Yes

Reply:
Fixed